### PR TITLE
fix: guest secured link password validation

### PR DESCRIPTION
**Pull request by e-maad**
## Description
[WPB-10395](https://wearezeta.atlassian.net/browse/WPB-10395)
Fixed form validation, added few tests and minor code cleanup

## Screenshots/Screencast (for UI changes)

https://github.com/user-attachments/assets/caff2d93-b1f5-44c9-a011-532492abf9bf

Edit: 
Screenshot 2024-08-08 at 12 14 24

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### Important details for the reviewers

(Delete this section if unnecessary)

- use (x) data
- can be reviewed commit-by-commit
- be sure to look at ...

[WPB-10395]: https://wearezeta.atlassian.net/browse/WPB-10395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/script/components/Modals/PrimaryModal/PrimaryModal.tsx [hunk 1]
```diff
@@ -307,10 +329,15 @@ export const PrimaryModalComponent: FC = () => {
                     type="password"
                     autoComplete="off"
                     value={passwordValue}
+                    ref={passwordValueRef}
                     onChange={event => setPasswordValue(event.currentTarget.value)}
+                    pattern={ValidationUtil.getNewPasswordPattern(Config.getConfig().NEW_PASSWORD_MINIMUM_LENGTH)}
+                    markInvalid={isFormSubmitted && !passwordValueRef.current?.validity.valid}
```
Comment (V-Gira) [inline src/script/components/Modals/PrimaryModal/PrimaryModal.tsx]: Question, it seems like other inputs have their border-color set to red when invalid, could you check why it's not the case here? We might need to add a last bit of css

File: src/script/components/Modals/PrimaryModal/PrimaryModal.tsx [hunk 2]
```diff
@@ -321,6 +348,7 @@ export const PrimaryModalComponent: FC = () => {
                     autoComplete="off"
                     value={passwordConfirmationValue}
                     onChange={event => setPasswordConfirmationValue(event.currentTarget.value)}
+                    markInvalid={isFormSubmitted && !checkGuestLinkPassword(passwordValue, passwordConfirmationValue)}
```
Comment (przemvs) [inline src/script/components/Modals/PrimaryModal/PrimaryModal.tsx]: You using this !checkGuestLinkPassword(passwordValue, passwordConfirmationValue) multiple times, please move this to variable :)

File: src/script/components/Modals/PrimaryModal/PrimaryModal.tsx (hunk 1 above)
Comment (przemvs) [inline src/script/components/Modals/PrimaryModal/PrimaryModal.tsx]: This You're using also multiple times ;) Please move to variable

Comment (e-maad) [reply src/script/components/Modals/PrimaryModal/PrimaryModal.tsx]: Nice catch 🦸 
There was some wrapper css that is overriding the default behavior. I realized we don't event need that css.
For the sake of consistency, we will add this css on UI toolkit if we still need it.

**Review by przemvs — APPROVED** [review]

**Pull request merged**